client/service/src/builder.rs

```diff
@@ -916,23 +916,23 @@ ServiceBuilder<
 			let is_validator = config.roles.is_authority();
 
 			let (import_stream, finality_stream) = (
-				client.import_notification_stream().map(|n| ChainEvent::NewBlock {
-					id: BlockId::Hash(n.hash),
-					header: n.header,
-					retracted: n.retracted,
-					is_new_best: n.is_new_best,
-				}),
+				client.all_blocks_notification_stream()
+					.map(|n| ChainEvent::BlockReceived {
+						header: n.header,
+						is_initial_sync: n.origin == BlockOrigin::NetworkInitialSync,
+						is_new_best: n.is_new_best,
+					}),
 				client.finality_notification_stream().map(|n| ChainEvent::Finalized {
 					hash: n.hash
 				})
 			);
 			let events = futures::stream::select(import_stream, finality_stream)
 				.for_each(move |event| {
 					// offchain worker is only interested in block import events
-					if let ChainEvent::NewBlock { ref header, is_new_best, .. } = event {
+					if let ChainEvent::BlockReceived { ref header, is_new_best, is_initial_sync, .. } = event {
 						let offchain = offchain.as_ref().and_then(|o| o.upgrade());
 						match offchain {
-							Some(offchain) if is_new_best => {
+							Some(offchain) if is_initial_sync || is_new_best => {
```

Comment:
Let's not change the behaviour yet, without changing the underlying `RuntimeApi`. Offchain Workers should be able to distinguish between the two cases, as most of them is not interested to run for every block.

```suggestion
							Some(offchain) if is_new_best && !is_initial_sync => {
```